headers.go

```diff
@@ -401,6 +407,16 @@ func ensureHeaderIV(h map[interface{}]interface{}) error {
 	return nil
 }
 
+// ensureNoCritical ensures crit parameter is not present in the header.
+//
+// Reference: https://datatracker.ietf.org/doc/html/rfc8152#section-3.1
+func ensureNoCritical(h map[interface{}]interface{}) error {
+	if hasLabel(h, HeaderLabelCritical) {
+		return errors.New("unprotected header: crit parameter must be places in the protected header")
```

Comment:
nit: `h` is just a header map from the point of view of this function.  The calling site should add the rest of the context.

```suggestion
		return errors.New("unexpected crit parameter found")
```